### PR TITLE
Add ctti package

### PR DESCRIPTION
**Pull request by caseymcc**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":
https://travis-ci.org/caseymcc/hunter/builds/414670150
https://ci.appveyor.com/project/caseymcc/hunter/build/1.0.172

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/275

* My change will resolve this "New package" request: https://github.com/ruslo/hunter/issues/1516

### CODE REVIEW

File: docs/packages/pkg/ctti.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.. spelling::
+
+    ctti
+
+.. index:: compiler ; ctti
+
+.. _pkg.ctti:
+
+OpenCL-cpp
```
Comment (ruslo) [inline docs/packages/pkg/ctti.rst]: `ctti`?

File: docs/packages/pkg/ctti.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+- Compilers: VS15, Clang >= 3.6.2
+- Does not support GCC with optimizations (https://github.com/Manu343726/ctti/issues/19)
+- Does not support VS17, bug in compiler
+ 
+
+
+Usage
+-----
+
+.. code-block:: cmake
```
Comment (ruslo) [inline docs/packages/pkg/ctti.rst]: Please use `.. literalinclude:: /../examples/*/CMakeLists.txt` here, see template file:
* https://raw.githubusercontent.com/ruslo/hunter/af6193b9a29ad9c53db4596faa9f1e4dd8502f2c/docs/packages/pkg/foo.rst

File: examples/ctti/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 3.0)
+
+# Emulate HunterGate:
+# * https://github.com/hunter-packages/gate
+include("../common.cmake")
+
+project(download_ctti)
+
+hunter_add_package(ctti)
```
Comment (ruslo) [inline examples/ctti/CMakeLists.txt]: Add `DOCUMENTATION_{START,END}`, see template file:
* https://github.com/ruslo/hunter/blob/af6193b9a29ad9c53db4596faa9f1e4dd8502f2c/examples/foo/CMakeLists.txt#L12-L18

**Comment by caseymcc** [comment]
should be good now

File: docs/packages/pkg/ctti.rst [hunk 4]
```diff
@@ -0,0 +1,26 @@
+.. spelling::
+
+    ctti
+
+.. index:: compiler ; ctti
+
+.. _pkg.ctti:
+
+ctti
+==========
```
Comment (ruslo) [inline docs/packages/pkg/ctti.rst]: Please use `====` here.

File: docs/packages/pkg/ctti.rst [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+-  `Official <https://github.com/Manu343726/ctti>`__
+-  `Hunterized <https://github.com/hunter-packages/ctti>`__
+-  `Example <https://github.com/ruslo/hunter/blob/master/examples/ctti/CMakeLists.txt>`__
+-  Added by `Casey <https://github.com/caseymcc>`__ (`pr-N <https://github.com/ruslo/hunter/pull/1518>`__)
```
Comment (ruslo) [inline docs/packages/pkg/ctti.rst]: Please update `pr-N`.

Comment (caseymcc) [reply docs/packages/pkg/ctti.rst]: didn't realize the N was th PR number.

**Comment by caseymcc** [comment]
Should be good this time. Thanks for the catches, maybe next time I should use the template rather than grabbing one of the other docs to start with.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.4330
* https://travis-ci.org/ingenue/hunter/builds/415495706?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.4331
* https://travis-ci.org/ingenue/hunter/builds/415509961?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.15

**Pull request merged**